.github/workflows/build.yml

```diff
@@ -25,10 +25,22 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
     - name: 'Install Dependencies'
-      run: sudo apt-get update && make ciprepare
+      run: |
+        sudo apt-get update && make ciprepare
+        sudo apt-get install libssl-dev
+        cargo install cargo-tarpaulin
     - name: 'Run rust tests'
-      run: make --keep-going test
+      run: |
+        make --keep-going test &&
+        cargo tarpaulin --out Xml --output-dir ./coverall
+    - name: 'upload the xml to coverall server'
+      uses: codecov/codecov-action@v2.1.0
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
This is not required for public repos as per the GitHub example at
https://crates.io/crates/cargo-tarpaulin